File: python/find_difference.py
class Solution(object):
    def findDifference(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[List[int]]
        """
        set1 = set(nums1)
        set2 = set(nums2)

        return [list(set1 - set2), list(set2 - set1)]

sol = Solution.new()
sol.findDifference([1, 2, 3], [2, 4, 6])

File: ruby/find_difference.rb
require "debug"

# Find the Difference of Two Arrays
# Given two 0-indexed integer arrays nums1 and nums2, return a list answer of size 2 where:

# answer[0] is a list of all distinct integers in nums1 which are not present in nums2.
# answer[1] is a list of all distinct integers in nums2 which are not present in nums1.
# Note that the integers in the lists may be returned in any order.

# Constraints:

# 1 <= nums1.length, nums2.length <= 1000
# -1000 <= nums1[i], nums2[i] <= 1000

def find_difference(nums1, nums2)
  [(nums1 - nums2).uniq, (nums2 - nums1).uniq]
end

p find_difference([1,2,3], [2,4,6])
p find_difference([1,2,3,3], [1,1,2,2])
